src/syrupy/extensions/amber/serializer.py

```diff
@@ -73,20 +74,23 @@ def read_file(cls, filepath: str) -> "SnapshotFossil":
         indent_len = len(cls._indent)
         snapshot_fossil = SnapshotFossil(location=filepath)
         try:
-            with open(filepath, "r", encoding="utf-8", newline="") as f:
+            with open(filepath, "r", encoding="utf-8", newline=None) as f:
                 test_name = None
                 snapshot_data = ""
                 for line in f:
                     if line.startswith(cls._marker_name):
-                        test_name = line[name_marker_len:-1].strip(" \r\n")
+                        test_name = line[name_marker_len:].strip(" \r\n")
                         snapshot_data = ""
                         continue
                     elif test_name is not None:
                         if line.startswith(cls._indent):
                             snapshot_data += line[indent_len:]
                         elif line.startswith(cls._marker_divider) and snapshot_data:
                             snapshot_fossil.add(
-                                Snapshot(name=test_name, data=snapshot_data[:-1])
+                                Snapshot(
+                                    name=test_name,
+                                    data=snapshot_data.rstrip(os.linesep),
```

Comment:
Consider the snapshot generated via:

```
assert snapshot == "apple\n"
```

Is there concern that we might strip newlines that are actually part of the data? Do we need to limit the rstrip to the length of the linesep? If there is concern, let's add a test for it (although I thought we did have one?).

Reply:
We already have tests that should cover it, basically the snapshot that gets generated is something like
```ambr
  '
  apple
  
  '
```
The last line with the `'` is what gets `rstrip`'d and that not part of the snapshot content to assert.